THREESCALE-8404: Add ACL and TLS support for Redis (#350)
3scale/apisonator: docs/configuration.md

- Applies to: listener, worker, cron.
- Format: string.

###  CONFIG_REDIS_USERNAME

- Redis ACL user name
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: string.

###  CONFIG_REDIS_PASSWORD

- Redis ACL password
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: string.

###  CONFIG_REDIS_CA_FILE

> - Certification authority certificate Redis should trust to accept TLS connections

Reviewer: ```suggestion
- Certification authority to validate Redis server TLS connections with
```

btw usually software also supports `CA_DIR` option but maybe for the use case we won't need. I haven't checked how we ensure trust with HTTPS connections that we use in 3scale.

Author: Yes, that's also supported by `redis-client`, I might implement it in the future... maybe...

Reviewer: The more important point of my comment is to figure out how the feature is to be used with the operator on-prem and make sure it suits :)

It's not presently clear to me, how are users expected to supplement this.

Author: Yes, it's good to provide a CA_DIR env var.

Reviewer: The dir thing may or may not be practical depending on the way users are supposed to supply that to the pod.

- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: path to file as string.

###  CONFIG_REDIS_CERT

- User certificate to connect to Redis through TLS
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: path to file as string.

> ###  CONFIG_REDIS_KEY

Reviewer: ```suggestion
###  CONFIG_REDIS_PRIVATE_KEY
```

I think private key is a more familiar term to describe what we have here. i.e. TLS certificate and a private key

> - User key to connect to Redis through TLS

Reviewer: >The path to the client SSL private key

- Applies to: listener, worker, cron.
- Format: string.

> ###  CONFIG_QUEUES_USERNAME

Reviewer: This I guess is for the `resque` background jobs. So I suggest naming it `CONFIG_RESQUE_USERNAME` or `CONFIG_JOBS_USERNAME`. This `QUEUES` word is a little ambiguous.

btw how nice that we have yet another background jobs implementation to care about...

Author: The `CONFIG_QUEUES_*` prefix was already there before, I prefer to not touch it.

Reviewer: hmm, I wondered how did you come up with this name.. I don't like it but what you say makes sense.

- Redis ACL user name
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: string.

###  CONFIG_QUEUES_PASSWORD

- Redis ACL password
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: string.

###  CONFIG_QUEUES_CA_FILE

- Certification authority certificate Redis should trust to accept TLS connections
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: path to file as string.

###  CONFIG_QUEUES_CERT

- User certificate to connect to Redis through TLS
- Optional. Defaults to empty.
- Applies to: listener, worker, cron.
- Format: path to file as string.

> ###  CONFIG_QUEUES_KEY

Reviewer: I wonder if all of these make more sense to have standard file locations. And users would be able to mount secrets at these locations with the file contents. Because these files will still need some way to be mounted into the containers anyway. So it would be double the work. Not sure, just a thought.

Author: I think it's good to have the variables, since they can be very useful for development: the standard file locations you mention would probably be out of the project folder. What we could do is adding a default in the case they are not provided.

Reviewer: This is what I meant.

Author: Done in https://github.com/3scale/apisonator/pull/350/commits/a81a4e9d2b45b49c6d8be418963de9dc2c03664e

Reviewer: will we have these variables ever empty? Just wondering how it will work. As if empty, I assume empty value may confuse us. Or not?

Author: I pushed some changes to ensure empty variables doesn't reach Redis: https://github.com/3scale/apisonator/pull/350/commits/c429e716213fe2ff7c9a66d0d838188f51c13ec4.
